crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -42,57 +45,119 @@ impl Ssa {
-            opcodes: Vec::new(),
-            return_witnesses: Vec::new(),
+    fn convert_ssa(mut self, ssa: Ssa) -> GeneratedAcir {
+        assert_eq!(
+            ssa.functions.len(),
+            1,
+            "expected only a single function to be present with all other functions being inlined."
+        );
+        let main_func = ssa.main();
+        let dfg = &main_func.dfg;
+        let entry_block = &dfg[main_func.entry_block()];
+
+        for param_id in entry_block.parameters() {
+            self.convert_ssa_block_param(*param_id, dfg);
         }
-    }
 
-    /// Converts an SSA value into a `AcirVar`
-    fn convert_ssa_value(
-        &mut self,
-        _value_id: &Id<Value>,
-        _dfg: &DataFlowGraph,
-    ) -> Result<AcirVar, AcirGenError> {
-        todo!()
+        for instruction_id in entry_block.instructions() {
+            self.convert_ssa_instruction(*instruction_id, dfg);
+        }
+
+        self.convert_ssa_return(entry_block.terminator().unwrap(), dfg);
+
+        self.acir_context.finish()
     }
 
-    /// Processes a binary operation and converts the result into an `AcirVar`
-    fn convert_ssa_binary(&mut self, _binary: &Binary, _dfg: &DataFlowGraph) -> AcirVar {
-        todo!()
+    /// Adds and binds an AcirVar for each numeric block parameter
+    ///
+    /// TODO: In the case of references, add AcirVars corresponding to the length of the array
+    /// and bind an ArrayId.
+    fn convert_ssa_block_param(&mut self, param_id: ValueId, dfg: &DataFlowGraph) {
+        let value = dfg[param_id];
+        let param_type = match value {
+            Value::Param { typ, .. } => typ,
+            _ => unreachable!("ICE: Only Param type values should appear in block parameters"),
+        };
+        match param_type {
+            Type::Numeric(..) => {
+                let acir_var = self.acir_context.add_variable();
+                self.ssa_value_to_acir_var.insert(param_id, acir_var);
+            }
+            Type::Reference => {
+                todo!("Alloc array for reference");
+            }
+            _ => unreachable!("ICE: invalid type {param_type} found in parameters"),
+        }
     }
 
-    /// Creates an `AcirVar` from an SSA value, if a corresponding `AcirVar` does not already
-    /// exist for it.
-    fn get_or_create_acir_var(&mut self, _value_id: &Id<Value>, _dfg: &DataFlowGraph) -> AcirVar {
-        todo!()
+    /// Converts an SSA instruction into its ACIR representation
+    fn convert_ssa_instruction(&mut self, instruction_id: InstructionId, dfg: &DataFlowGraph) {
+        let instruction = &dfg[instruction_id];
+        match instruction {
+            Instruction::Binary(binary) => {
+                let result_acir_var = self.convert_ssa_binary(binary, dfg);
+                let result_value_id = dfg
+                    .instruction_results(instruction_id)
+                    .first()
+                    .expect("Binary ops have a single result");
```

Comment:
This message is not entirely correct; This will trigger if there are no results, not if there is only one result.

If you want to ensure there is only one result, you need to do an assert operation on the instruction_results() to be 1.

Reply:
changed the length assert